packages/api-client/src/v2/hooks/use-document-watcher.ts

```diff
@@ -0,0 +1,72 @@
+import type { WorkspaceStore } from '@scalar/workspace-store/client'
+import { type MaybeRefOrGetter, computed, toValue, watch } from 'vue'
+
+/**
+ * Watches the specified document in the workspace store and periodically rebases it with its remote source.
+ *
+ * This utility sets up a watcher on the given document. If the document specifies an 'x-scalar-original-source-url',
+ * this hook polls the remote source every second (using setInterval) and calls `store.rebaseDocument`.
+ * If rebase conflicts are detected, it applies automatic conflict resolution by preferring remote changes.
+ *
+ * @param params - Object with:
+ *   - documentName: the name/key of the document to watch and rebase.
+ *   - store: the WorkspaceStore instance.
+ *
+ * @example
+ * ```ts
+ * import { useDocumentWatcher } from '@/hooks/use-document-watcher'
+ * import { useScalarWorkspaceStore } from '@scalar/workspace-store'
+ *
+ * const store = useScalarWorkspaceStore()
+ * useDocumentWatcher({ documentName: 'myApi', store })
+ * ```
+ */
+export const useDocumentWatcher = ({
+  documentName,
+  store,
+  timeout = 5000,
+}: {
+  documentName: MaybeRefOrGetter<string>
+  store: WorkspaceStore
+  timeout?: number
+}) => {
+  let initialTimeout = timeout
+  const document = computed(() => store.workspace.documents[toValue(documentName)])
+
+  let interval: NodeJS.Timeout | null = null
```

Comment:
```suggestion
  let interval: ReturnType<typeof setInterval> | null = null
```
It doesn't matter too much here, but we probably shouldn't use nodejs types on the front-end